src/ssl.c

```diff
@@ -12005,6 +12005,166 @@ static int wolfSSL_remove_ciphers(char* list, int sz, const char* toRemove)
+        }
+    }
+    
+    len += listsz + 2;
+    
+    /* build string */
+    if (len > 0) {
+        locallist = (char*)XMALLOC(len, ctx->heap,
+                                           DYNAMIC_TYPE_TMP_BUFFER);
+        /* sanity check */
+        if (!locallist)
+            return NULL;
+        
+        XMEMSET(locallist, 0, len);
+        
+        head = locallist;
+        
+        if (!onlytlsv13suites)
+        {
+            /* always tls13 suites in the head position */
+            XSTRNCPY(locallist, list, len);
+            locallist += listsz;
+            *locallist++ = ':';
+            *locallist = 0;
+            len -= listsz + 1;
+        }
+        
+        for(idx = 0; idx < suites->suiteSz; idx++) {
+            cipherSuite0 = suites->suites[idx];
+            cipherSuite  = suites->suites[++idx];
+            
+            if (onlytlsv13suites && cipherSuite0 == TLS13_BYTE) {
+                enabledcs = GetCipherNameInternal(cipherSuite0, cipherSuite);
+            } else if (!onlytlsv13suites && cipherSuite0 != TLS13_BYTE) {
+                enabledcs = GetCipherNameInternal(cipherSuite0, cipherSuite);
+            } else
+                continue;
+            
+            ianasz = (int)XSTRLEN(enabledcs);
+            if (ianasz + 1 < len) {
+                XSTRNCPY(locallist, enabledcs, len);
+                locallist += ianasz;
+                
+                *locallist++ = ':';
+                *locallist = 0;
+                len -= ianasz + 1;
+            }
+            else{
+                XFREE(locallist, ctx->heap, DYNAMIC_TYPE_TMP_BUFFER);
+                return NULL;
+            }
+        }
+        
+        if (onlytlsv13suites) {
+            XSTRNCPY(locallist, list, len);
+            locallist += listsz;
+            *locallist = 0;
+        }
+        
+        return head;
+    } else 
+        return NULL;
+}
+
+/*                                                                          */
+/* check if the list has TLS13 and pre-TLS13 suites                         */
+/* @param list cipher suite list that user want to set                      */
+/* @return mixed: 0, only pre-TLS13: 1, only TLS13: 2                       */
+static int CheckcipherList(const char* list)
+{
+    int findTLSv13Suites = 0;
+    int findbeforeSuites = 0;
+    const int suiteSz = GetCipherNamesSize();
+    const CipherSuiteInfo* names = GetCipherNames();
+    
+    char* next = (char*)list;
+    
+    do {
+        char*  current = next;
+        char   name[MAX_SUITE_NAME + 1];
+        int    i;
+        word32 length;
+        
+        next   = XSTRSTR(next, ":");
+        length = min(sizeof(name), !next ? (word32)XSTRLEN(current) /* last */
+                                         : (word32)(next - current));
+        XSTRNCPY(name, current, length);
+        name[(length == sizeof(name)) ? length - 1 : length] = 0;
+        
+        for (i = 0; i < suiteSz; i++) {
```

Comment:
Use wolfSSL_get_cipher_suite_from_name().

Reply:
Changed to use wolfSSL_get_cipher_suite_from_name()